src/App.js

```diff
@@ -18,25 +19,30 @@ function App() {
       <header>
         <h1>Todo List</h1>
       </header>
-      <Form
-        todos={todos}
-        setTodos={setTodos}
-        action={action}
-        setAction={setAction}
-        itemToEdit={itemToEdit}
-        setFilteredTodos={setFilteredTodos}
-        setInputText={setInputText}
-        inputText={inputText}
-      />
-      <ToDoList
-        todos={todos}
-        setTodos={setTodos}
-        filteredTodos={filteredTodos}
-        action={action}
-        setAction={setAction}
-        setItemToEdit={setItemToEdit}
-        setInputText={setInputText}
-      />
+      <AppContext.Provider
+        value={{
+          setTodos,
+          action,
+          setAction,
+          itemToEdit,
+          setInputText,
+          inputText,
+        }}
+      >
+        <Form setFilteredTodos={setFilteredTodos} todos={todos} />
+      </AppContext.Provider>
+      <AppContext.Provider
```

Comment:
Usually you wont need separating Context Providers per different components, you should just wrap it once at the parent or highest app scope (App.js), thus all components will have access to it. Separating contexts could be a solution for totally different contexts (ie `        <ContextOne.Provider value={contextOneVal}>
          <ContextTwo.Provider value={contextTwoVal}>
                <div>App Here</div>
          </ContextTwo.Provider>
        </ContextOne.Provider>`). Afais you are just trying to pass only needed props, in that case you should know, that besides context provider, context consumer exists, that is filtering all props.

Reply:
fixed